core/src/main/java/org/apache/calcite/plan/SubstitutionVisitor.java

```diff
@@ -1899,8 +1899,12 @@ public static MutableRel unifyAggregates(MutableAggregate query,
         if (i < 0) {
           return null;
         }
+        SqlAggFunction aggFunction = getRollup(aggregateCall.getAggregation());
```

Comment:
How about some doc to explain why matching should fail when the function doesn't support roll-up.

Reply:
Mark it as `final`